### PR TITLE
Update documentation about parsing enums in TSV/CSV

### PR DESCRIPTION
**Pull request by Avogar**
Changelog category (leave one):
- Documentation (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Update documentation about parsing enums in TSV/CSV according to https://github.com/ClickHouse/ClickHouse/pull/16834. 

Detailed description / Documentation draft:
...

### CODE REVIEW

File: docs/en/interfaces/formats.md [hunk 1]
```diff
@@ -138,6 +138,9 @@ empty input fields are replaced with default values. For complex default express
 
 Each element of [Nested](../sql-reference/data-types/nested-data-structures/nested.md) structures is represented as array.
 
+In input data ENUM values can be represented as names or as ids. First we try to match input value to ENUM name, if we fail and input value is a number, then we try to match this number to ENUM id.
+If input data contains only ENUM ids, it's recommended to enable setting [input_format_tsv_enum_as_number](../operations/settings/settings.md#settings-input_format_tsv_enum_as_number) to optimize ENUM parsing.
+
```
Comment (FArthur-cmd) [inline docs/en/interfaces/formats.md:143]:
```suggestion
In input data, ENUM values can be represented as names or as ids. First, we try to match the input value to the ENUM name. If we fail and the input value is a number, we try to match this number to ENUM id.
If input data contains only ENUM ids, it's recommended to enable the setting [input_format_tsv_enum_as_number](../operations/settings/settings.md#settings-input_format_tsv_enum_as_number) to optimize ENUM parsing.

```

Comment (FArthur-cmd) [reply docs/en/interfaces/formats.md:143]: same for lines 414-416

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -447,7 +447,17 @@ When the `input_format_tsv_enum_as_number` setting is enabled:
 ```sql
 SET input_format_tsv_enum_as_number = 1;
 INSERT INTO table_with_enum_column_for_tsv_insert FORMAT TSV 102	2;
-INSERT INTO table_with_enum_column_for_tsv_insert FORMAT TSV 103	1;
+INSERT INTO table_with_enum_column_for_tsv_insert FORMAT TSV 103	'first';
+```
+
+throws an exception.
```
Comment (FArthur-cmd) [inline docs/en/operations/settings/settings.md:453]: Maybe it is better to separate lines to show that the first works correctly and the second throws an exception. Or make comments for lines, or write more information after example. (this is just a thought. If you think that it is obvious, you can ignore it)

Comment (FArthur-cmd) [reply docs/en/operations/settings/settings.md:453]: Same for CSV

**Review by FArthur-cmd — APPROVED** [review]

**Comment by ka1bi4** [comment]
Internal documentation ticket: DOCSUP-18084

**Pull request merged**